README.md

```diff
@@ -200,6 +201,19 @@ In case of success, return all brokers and all partitions of all topics.
 In case of errors, returns `nil` with a string describing the error.
 
 
+[Back to TOC](#table-of-contents)
+
+#### choose_api_version
+
+`syntax: api_version = c:choose_api_version(api_key, min_version, max_version)`
+
+This helps the client to select the correct version of the `api_key` corresponding to the API.
+
+When `min_version` and `max_version` are provided, it will act as a limit and the selected versions in the return value will not exceed their limits no matter how high or low the broker supports the API version. When they are not provided, it will follow the range of versions supported by the broker.
```

Comment:
the `min_version` and `max_version` are const for this library, right?
I'm not clear about how do the users will use this new API.
Could you please add a sample for it, thanks.

Reply:
@doujiang24 In fact this API is not used by the user, it will be used in the producer or consumer implementation, and through this API we can choose an available version by combining the supported version of the broker with the supported version in our client. 

------

Let me give you an example:

We have a set of brokers that support the Fetch API version 2 to 13, and in the client side we have implemented support for version 0 to 8. At this point it will automatically select the highest version supported on the client side, i.e. version 8, instead of broker's version 13. **The selection rule is the highest version in the overlap of the client support range and broker support range.**

```text
client:  0  1  2  3  4  5  6  7  8
broker:        2  3  4  5  6  7  8  9  10  11  12
overlap:       2  3  4  5  6  7  8
max of overlap:                  8
```

Usage pseudocode:
```text
-- consumer.lua

-- get the version of Fetch API and also mark that our client supports version 0 to 8.
api_version = choose_api_version(FetchAPI, 0, 8)

-- create Fetch Req data
req = req_new(FetchAPI, api_version)
```